src/utils/checkTypes.js

```diff
@@ -1,10 +1,12 @@
 import invariant from 'invariant'
 import isPlainObject from './isPlainObject'
 
-function typecheck(type, name, obj) {
+function typecheck(types, name, obj) {
   invariant(
-    typeof obj === type,
-    `${name} must be a ${type}. Instead received a %s.`,
+    Array.isArray(types)
+      ? types.some(t => typeof obj === t)
+      : typeof obj === types,
+    `${name} must be a ${types}. Instead received a %s.`,
```

Comment:
This message will be weird if `${types}` is an array. Probably need to do some `join`ing.  Also, it might be easier to convert `types` to an `Array` before the `invariant` to avoid having two special cases in here.